# make an own build.xml because the sources do not contain all build files
--- /dev/null
+++ b/build.xml
@@ -0,0 +1,39 @@
+<project name="chatty" default="jar">
+ <target name="clean">
+   <delete dir="bin"/>
+   <delete dir="dist"/>
+ </target>
+
+ <target name="compile" depends="clean">
+    <mkdir dir="bin"/>
+    <javac srcdir="src/chatty" encoding="utf-8" destdir="bin">
+       <classpath>
+          <fileset dir="assets/lib">
+          	 <include name="slf4j-api-1.7.29.jar" />
+          	 <include name="Java-WebSocket-1.4.0.jar" />
+             <include name="jintellitype-1.3.8.jar" />
+             <include name="JTattoo-1.6.12-1.jar" />
+             <include name="txtmark-0.13.jar" />
+             <include name="tyrus-standalone-client-1.13.1.jar" />
+          </fileset>
+       </classpath>
+    </javac>
+    <copy todir="bin">
+      <fileset dir="src" excludes="**/*.java"/>
+    </copy>
+ </target>
+
+ <target name="jar" depends="compile">
+    <mkdir dir="dist"/>
+    <jar manifest="manifest.mf" destfile="dist/chatty.jar">
+       <fileset dir="bin" />
+        <zipgroupfileset dir="assets/lib">
+        	 <include name="slf4j-api-1.7.29.jar" />
+        	 <include name="Java-WebSocket-1.4.0.jar" />
+             <include name="JTattoo-1.6.12-1.jar" />
+             <include name="txtmark-0.13.jar" />
+             <include name="tyrus-standalone-client-1.13.1.jar" />
+        </zipgroupfileset>
+    </jar>
+ </target>
+</project>
